Source/CubicHorndeski/CubicHorndeski.impl.hpp

```diff
@@ -55,33 +69,49 @@ CubicHorndeski<coupling_and_potential_t>::compute_Sij_TF_and_S(
     const diff2_vars_t<Tensor<2, data_t>> &d2, const vars_t<data_t> &advec,
     const Coordinates<data_t> &coords) const
 {
-    expressions<data_t> E;
+    UsefulQuantities<data_t> quantities;
     using namespace TensorAlgebra;
     const auto h_UU = compute_inverse_sym(vars.h);
     const auto chris = compute_christoffel(d1.h, h_UU);
 
-    pre_compute_no_gauge(E, vars, d1, d2, h_UU, chris.ULL);
+    compute_useful_quantities(quantities, vars, d1, d2, h_UU, chris.ULL);
 
     SijTFAndS<data_t> out;
 
+    // dphi_dot_dphi - not conformal
+    data_t dphi_dot_dphi = 0.;
+    FOR(i, j)
+    {
+        dphi_dot_dphi += vars.chi * h_UU[i][j] * d1.phi[i] * d1.phi[j];
+    }
+
+    // X
+    data_t X = 0.5 * (vars.Pi * vars.Pi - dphi_dot_dphi);
+
     // compute potential and add contributions to EM Tensor
     const data_t chi_regularised = simd_max(vars.chi, 1e-6);
 
+    data_t repeated_quantity =
```

Comment:
Hmm... why repeated? If we can't think of a good name we could call it "another_useful_quantity", which would be funny (I think so anyway) :-)

Reply:
Okay I'll put it! I didn't come up with any better name hahah